lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -578,10 +439,79 @@ def _partition_pdf_batched_named(
     def _run_pdfminer(pdf_path, hash_key, use_cache):
         pdfminer = PDFMinerExtractor()
         with LogTime("pdfminer_extract", log_start=True):
-            pdfminer_layout = pdfminer.extract(pdf_path, hash_key, use_cache)
+            pdfminer_layout = pdfminer.extract_document(pdf_path, hash_key, use_cache)
 
         return pdfminer_layout
 
+    @requires_modules("easyocr", extra="local-inference")
+    def process_batch(
+        self,
+        batch: list[Image.Image],
+        pdf_generator,
+        threshold,
+        use_ocr,
+        ocr_images,
+        ocr_tables,
+        extract_table_structure,
+        table_structure_extractor,
+        extract_images,
+        use_cache,
+    ) -> Any:
+        import easyocr
+
+        self._init_model()
+
+        with LogTime("infer"):
+            assert self.model is not None
+            deformable_layout = self.model.infer(batch, threshold, use_cache)
+
+        gc_tensor_dump()
+        assert len(deformable_layout) == len(batch)
+        if use_ocr:
+            with LogTime("ocr"):
+                if self.ocr_table_reader is None:
+                    self.ocr_table_reader = easyocr.Reader(["en"])
+
+                extract_ocr(
+                    batch,
+                    deformable_layout,
+                    ocr_images=ocr_images,
+                    ocr_tables=ocr_tables,
+                    table_reader=self.ocr_table_reader,
+                )
+        else:
+            pdfminer_pages = []
+            for _ in range(len(batch)):
+                pdf_page = pdf_generator.__next__()
+                pdfminer_pages.append(self.pdf_extractor.extract_page(pdf_page))
+
+            assert len(pdfminer_pages) == len(deformable_layout)
+            with LogTime("pdfminer_supplement"):
+                for d, p in zip(deformable_layout, pdfminer_pages):
+                    self._supplement_text(d, p)
+        # else pdfminer happens in parent since it is whole document.
+        if extract_table_structure:
+            with LogTime("extract_table_structure_batch"):
+                if table_structure_extractor is None:
+                    table_structure_extractor = DEFAULT_TABLE_STRUCTURE_EXTRACTOR(device=self.device)
```

Comment:
minor, and we can maybe leave this to another cleanup PR, but we already initialize this in _extract_pdf_batched as well. Seems redundant to do it both places.

Reply:
Makes sense, I've removed it